heim-runtime/src/linux.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(target_os = "linux")]
+
+/// Linux-specific extension for configuring custom paths for Procfs
+/// and Sysfs.
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+use std::path::{Path, PathBuf};
+
+/// Holds global state for custom paths. Static's leveraging this type
+/// can only be set ONCE per binary run.
+type RootLock = OnceCell<PathBuf>;
```

Comment:
Not sure if "lock" makes sense here after all the changes. Maybe `RootPathCell`?